### ISSUE TITLE
Any way to access my signature in Preview?

### ISSUE THREAD

**Issue by mfilej** [roeybiran/finbar-issues#22]
Hi, this is the only case where I still use the native macOS menu bar search interface.

I have a saved signature in Preview and it doesn't show up in Finbar.

I saw posts online saying that this particular menu item does not expose any accessibility features, so I wouldn't be surprised if it is not possible for Finbar to do this at all.

But I wanted to check anyway, because it's been driving me nuts for a while.

tl;dr Can I use Finbar to get to the menu shown in the screenshot below?

![CleanShot 2024-11-05 at 17 13 20@2x](https://github.com/user-attachments/assets/39ef6308-2b95-4d6a-b685-f56fab49ccae)

![CleanShot 2024-11-05 at 17 16 44@2x](https://github.com/user-attachments/assets/f1108d58-e58e-4e82-a20f-4c16f3653a84)

**Comment by roeybiran** [comment]
Hi!

Thanks for the feedback and for using the issue tracker. Glad you like Finbar!

I checked and indeed those menu items don’t have a title, which prevents Finbar from indexing them. There are 2 possible workarounds:

1. Simply giving each of those menu items the alias “Signature” — like macOS does. This would entail an update on my part.
2. Leverage Finbar’s ability to run custom scripts. For example, should you want to click the first signature in that sub-menu, you'd add the following Preview script. 
```bash
osascript -e 'tell app "System Events" to click menu item 1 of menu 1 of menu item "Signature"  of menu 1 of menu item "Annotate"  of menu 1 of menu bar item "Tools"  of menu bar 1 of application process "Preview"'
```

And if it’s second signature that you’d like to click, change the first `1` in the script to `2`, and so on. Then, save it under `~/Library/Application Scripts/com.roeybiran.Finbar/com.apple.Preview`, and hit “Reload Scripts” in Finbar’s menu. After selecting this item, Finbar will add it to your recents just like a standard menu item.

> 
> I saw posts online saying that this particular menu item does not expose any accessibility features, so I wouldn't be surprised if it is not possible for Finbar to do this at all.
> 

I’d like to read those too, could you please share?

Roey

**Comment by mfilej** [comment]
Thanks so much for the thoughtful explanation!

The solution with the script works wonders, don't know why I haven't thought about it.

For anyone else looking to adopt this, I had to wrap the above snippet in `tell application "System Events"` and `end tell` (and use simple quotes around `"Preview"`).

Here is a post discussing the lack of an AXTitle: https://apple.stackexchange.com/a/428966

Feel free to close this issue.

**Comment by roeybiran** [comment]
Oops! Totally forgot about the `tell app "System Events"` part, and also Safari messed up my quotes. Edited.

Anyway, I saw that those menu items expose an `AXDescription` property with a value of `Signature`, and the idea of falling back to this value whenever `AXTitle` is absent is worth investigating. Needs some testing because it might have some unintended side effects.

Thank you!
